core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -137,6 +138,22 @@ trait EntityDecoderInstances {
     EntityDecoder.decodeBy(MediaRange.`*/*`)(collectBinary)
   }
 
+  implicit def sourceDecoder[A](implicit W: EntityDecoder[A]): EntityDecoder[Process[Task,A]] =
+    new EntityDecoder[Process[Task,A]] {
+      override def decode(msg: Message): DecodeResult[Process[Task,A]] = {
+        val result = msg.body.flatMap{ byteVector =>
+          val fakeMessage = Request(body = Process.emit(byteVector))
+          Process.await(W.decode(fakeMessage).run) {
```

Comment:
I'm inclined to agree with your comment. Sorry, it seems to say basically the same thing as what I noted. :smile:

A websocket client shouldn't be that difficult to make, but I think we should discuss it in an issue before getting to to deep into code. I have a few ideas of what it would look like to me, but no code. Feel free to start an issue about it.